lib/drivers/abstract/spellbook.js

```diff
@@ -109,8 +109,11 @@ function formatLiteral(spell, ast) {
 
   if (value == null) {
     return 'NULL';
+  } else if (Array.isArray(value)) {
+    if (value.length) return `(${value.map(() => '?').join(', ')})`;
+    return '';
```

Comment:
What bug was this change intend to fix?

Reply:
Empty array in where conditions may cause SQL error
```js

User.where({
  id: [ ],
});
/**
 * error: SELECT * FROM `users` WHERE `deleted_at` is NULL AND `id` in ()
 * correct: SELECT * FROM `users` WHERE `deleted_at` is NULL 
 * */
```